core/src/test/scala/unit/kafka/zk/KafkaZkClientTest.scala

```diff
@@ -194,7 +194,56 @@ class KafkaZkClientTest extends ZooKeeperTestHarness {
 
     zkClient.createTopicAssignment(topic2, secondAssignment)
 
-    assertEquals(Set(topic1, topic2), zkClient.getAllTopicsInCluster.toSet)
+    assertEquals(Set(topic1, topic2), zkClient.getAllTopicsInCluster())
+  }
+
+  @Test
+  def testGetAllTopicsInClusterTriggersWatch(): Unit = {
+    zkClient.createTopLevelPaths()
+    val latch = registerChildChangeHandler(1)
+
+    // Listing all the topics and register the watch
+    assertTrue(zkClient.getAllTopicsInCluster(true).isEmpty)
+
+    // Verifies that listing all topics without registering the watch does
+    // not interfere with the previous registered watcher
+    assertTrue(zkClient.getAllTopicsInCluster(false).isEmpty)
+
+    zkClient.createTopicAssignment(topic1, Map.empty)
+
+    assertTrue("Failed to receive watch notification",
+      latch.await(100, TimeUnit.MILLISECONDS))
```

Comment:
In this case, normally the test will complete well below the timeout. So, we could set the timeout to be sth more generous like 5 secs to avoid flakiness.

In the other cases where the test always wait for the timeout, then we want to set the timeout to a much smaller value to control the testing time.

Reply:
Right. My mistake. I put them on par while trying various timeouts in order to avoid putting a value too low. I did not pay attention that I forgot to change them back.